crates/astria-sequencer/src/app/mod.rs

```diff
@@ -338,8 +408,11 @@ impl App {
         // included in the block
         let res = generate_rollup_datas_commitment(&signed_txs_included, deposits);
 
+        // inject the extended commit info into the start of the block's txs
+        let mut txs = vec![extended_commit_info_bytes.into()];
+        txs.extend(res.into_transactions(included_tx_bytes));
```

Comment:
Not a big deal, but we can avoid the `mut` variable and potentially two allocations by doing:
```suggestion
        let txs = std::iter::once(extended_commit_info_bytes.into())
            .chain(res.into_transactions(included_tx_bytes))
            .collect();
```
Or maybe `GeneratedCommitments::into_transactions` should take the `extended_commit_info_bytes` as an arg and we can just add it to the front of the Vec inside there?

Reply:
nice, that's a lot better!